packages/date-picker/src/DatePicker/components/Calendar/Calendar.tsx

```diff
@@ -25,76 +28,68 @@ export type CalendarProps = {
   }
   classNameOverride?: string
   value?: Date
-  initialMonth?: Date
-  firstDayOfWeek: number
-  disabledDays?: Modifier | Modifier[]
-  onDayChange: (
-    day: Date,
-    modifiers: DayModifiers,
-    e: React.MouseEvent<HTMLDivElement, MouseEvent>
-  ) => void
-  range?: boolean
-  selectedRange?: RangeModifier
-  modifiers?: RangeModifier
-  onKeyDown?: (e: React.KeyboardEvent) => void
+  defaultMonth?: Date
+  weekStartsOn: 0 | 1 | 2 | 3 | 4 | 5 | 6 | undefined
```

Comment:
Ah, the culprit. We should have this accept the enum and do the type guard. Then the DatePicker and DateRangePicker can both just happily use the enum.